metadata/metadata.go

```diff
@@ -0,0 +1,130 @@
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"github.com/mackerelio/mackerel-agent/config"
+	"github.com/mackerelio/mackerel-agent/logging"
+)
+
+var logger = logging.GetLogger("metadata")
+
+// Generator generates metadata
+type Generator struct {
+	Name         string
+	Config       *config.MetadataPlugin
+	Tempfile     string
```

Comment:
[nit] I think `Tempfile` is not a best naming. Although it may be created in tmpdir, but It is not "temporary" file, and exists for a long time. If you have good idea, rename this field name.